### DISCUSSION TITLE
Iusse in translating the guestbook example to vb.net

### DISCUSSION THREAD

**Discussion by ghost** [canton7/Stylet#291] [Q&A]
hello, I'm traslating the guestbook example in vb.net; I followed the instructions, but I get this problem: 
In the GuestBookViewModel.cs when  I try to  expose the properties of a model that can notify changes, I get BC31429 'lastName' is ambiguous because there are different types of members with the same name in class. And that for both firstname and last name. Any advice please?

**Comment by canton7** [comment]
Where did you find a GuestBookViewModel? I don't have a sample guest book project in the repository.

**Reply by ghost** [reply]
here https://github.com/canton7/Stylet/wiki/Introductory-example:-a-guest-book-application

**Reply by canton7** [reply]
Oh, right. That was contributed by a third party (who I don't think is prepared to support or maintain it).

What vb code do you have, which doesn't compile?

**Comment by ghost** [comment]
Thanks, I think I solved, It was my lack on c# and I done a   bad translation.
I would ask another thing:
I read here https://github.com/canton7/Stylet/wiki/Actions#actions-and-styles  about style .
Would this mean that If I want use Action I can't use skin, or usercontrol like that? https://stackoverflow.com/questions/42855236/wpf-custom-led-checkbox
I'm doing some test and seem not work, but I've not understand if is my fault or if I'm trying something that is not allowed.
Sorry if it's stupid question, but I'm don't known a lot about wpf and mvvm, I'm learning, I'm a winform guy..

**Reply by canton7** [reply]
Cool!

You can't use an action in a Style's setter, i.e. `<Setter Property="..." Value="{s:Action...}"/>`

**Comment by ghost** [comment]
ah ok! thanks!

**Comment by ghost** [comment]
I'd like to use the sincfusion docking manager control togheter Stylet, so I'm studying both documentation; 
here some link
https://help.syncfusion.com/wpf/docking/pattern-and-practices
https://www.syncfusion.com/blogs/post/mvvm-adapter-for-wpf-docking-manager.aspx

I found doc and code examples for many mvvm framework(caliburnmicro,prism,mvvmlight), but not for Stylet. is it the way showed in the syncfusion docs suitable for stylet or there is a better approach?

**Reply by canton7** [reply]
Stylet is pretty similar for Caliburn.Micro, so the same general approaches should apply.

**Comment by ghost** [comment]
Thanks

**Comment by ghost** [comment]
Hi, I'm trying to do bind some usercontrol to display them in a content control as TimCorey did in this video with CM: https://www.youtube.com/watch?v=laPFq3Fhs8k&t=3974s
my solution is like:
![immagine](https://user-images.githubusercontent.com/82973069/139011680-6d95d607-49eb-47c1-a1ae-d445147740ff.png)
my rootView :
```
<Window x:Class="RootView"
        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
         xmlns:s="https://github.com/canton7/Stylet"
        xmlns:local="clr-namespace:TestStylet"
          xmlns:syncfusion="http://schemas.syncfusion.com/wpf"
           xmlns:syncfusionskin ="clr-namespace:Syncfusion.SfSkinManager;assembly=Syncfusion.SfSkinManager.WPF"
        syncfusionskin:SfSkinManager.VisualStyle="Office2019Black"
        xmlns:myLed="clr-namespace:myLedCore;assembly=myLedCore"
        mc:Ignorable="d"
        Title="RootView" 
        Height="300" Width="300"    >
    <Grid>
        <TextBlock>Hello, World</TextBlock>
        <Button Command="{s:Action ShowA}" Margin="50,50,50,172"  syncfusionskin:SfSkinManager.VisualStyle="Office2019Black"  >Click me</Button>
        <ContentControl Content="{Binding UserControl1ViewModel}"  HorizontalAlignment="Left" Height="106" Margin="50,142,0,0" VerticalAlignment="Top" Width="185" />
    </Grid>
       </Window>
```
and RootViewModel
```
Imports Stylet

Public Class RootViewModel
    Inherits Conductor(Of Object)
    Private windowManager As IWindowManager

    Public Sub New(ByVal windowManager As IWindowManager)
        Me.windowManager = windowManager
    End Sub

    ' Public Sub ShowAWindow()
    'Dim viewModel = New DisplayedAsWindowViewModel()
    ' Me.windowManager.ShowWindow(viewModel)
    ' End Sub
    ' Public Sub DoSomething(argument As String)

    ' ShowAWindow()

    'End Sub
    Public Sub ShowA()
        Me.ActivateItem(New UserControl1ViewModel)
    End Sub

End Class
```
But I can't make it work. What am I missing?

**Reply by canton7** [reply]
Do you ever call `ShowA`?

**Comment by ghost** [comment]
I call it  in xaml : <Button Command="{s:Action ShowA}"

**Reply by canton7** [reply]
Right, and what do you see which makes you think that it isn't working?

**Comment by ghost** [comment]
I did not get exception, but the usercontrol is not showed; I get the warning "member not found in data context RootViewModel" at the line:  <ContentControl Content="{Binding UserControl1ViewModel}" ...

**Reply by canton7** [reply]
Right, you need to bind to ActiveItem. Please read the docs page on screens and conductors

**Reply by ghost** [reply]
I get i work. Thanks. About screens and conductors, if I correctly understand, if I one window, and let's say 4 usercontrol, that I want to display in 4 portion(content controls) of the main window (they should be all active right?), I should to use "Conductor<T>.Collection.AllActive" ?

**Reply by canton7** [reply]
You could, or you could simply make those 4 usercontrols proeprties on your main window's VM.

```cs
public class MainViewModel : Screen
{
    public Child1ViewModel Child1 { get; }
    // ...

    public MainViewModel()
    {
        Child1 = ...
    }
}
```

```xaml
<ContentControl s:View.Model="{Binding Child1}"/>
```

**Reply by ghost** [reply]
> need to bind to ActiveItem. Please read the docs page on screens and conductors
I've lost the example that I manage to work some time ago, and now I'm facing similar problem:
I binded to ActiveItem `<ContentControl s:View.Model="{Binding Child1}"/>`
When I run the app and click the button, the usercontrol is not showed, but instead a string with the name of the viewmodel. What am I missing?
